### DISCUSSION TITLE
Inconsistent PNG file size when manipulating the MagickImage.Quality

### DISCUSSION THREAD

**Discussion by renatoatfugro** [dlemstra/Magick.NET#1589] [Help] [answered]
Hi all!

I'm working on a small app where the user can control certain aspects of their image, such as format (Jpeg/Png), Quality (1-100), Color Mode (RGB, GrayScale) and Bit Depth (2-bit, 4-bit and 8-bit options).

The user feeds a Bitmap file, and based on the provided values mentioned above I write the changes to a memory stream print the array length it to inform the user about the estimated size of the output image:

```
var bitmap = new Bitmap(@"C:\temp\image.bmp");
using var tempStream = new MemoryStream();
bitmap.Save(tempStream, ImageFormat.Bmp);

var image = new MagickImage(tempStream.ToArray())
{
    Format = MagickFormat.Png,
    Quality = 100
};
image.SetBitDepth(8);
image.Grayscale();
using var stream = new MemoryStream();
image.Write(stream);
Console.WriteLine($"Png, Quality 100, Size {stream.ToArray().Length} bytes");
```

I noticed that this code using `Format = MagickFormat.Jpeg` gives me a consistent decrease/increase of the file size according to the `Quality` parameter:
```
Jpeg, Quality 92, Size 176554 bytes
Jpeg, Quality 93, Size 187915 bytes
Jpeg, Quality 94, Size 201850 bytes
Jpeg, Quality 95, Size 217083 bytes
Jpeg, Quality 96, Size 235019 bytes
```

However, with Png, it behaves strangely:
```
Png, Quality 92, Size 366245 bytes
Png, Quality 93, Size 366245 bytes
Png, Quality 94, Size 366245 bytes
Png, Quality 95, Size 297701 bytes
Png, Quality 96, Size 297701 bytes
```

I'm looking for options for decreasing the size of Png files, and through Quality does not seem to be correct way.

Any suggestions?

**Comment by dlemstra** [comment, marked as answer]
The quality works different for PNG files. You can find more information here: https://www.imagemagick.org/script/command-line-options.php#quality.

**Reply by renatoatfugro** [reply]
Hi @dlemstra, thank you for your response! this does clarify my doubts regarding the differences in image size.
so, in the sample I provided, the values for filter-type and compression-level would be:
```
quality: 92 filter-type: 2 compression-level: 9
quality: 93 filter-type: 3 compression-level: 9
quality: 94 filter-type: 4 compression-level: 9
quality: 95 filter-type: 5 compression-level: 9
quality: 96 filter-type: 6 compression-level: 9
```
which makes complete sense, given that up to filter-type = 4 we have different strategies applied.
once again, thank you!
